parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -542,6 +606,10 @@ public static Binary fromCharSequence(CharSequence value) {
     return new FromCharSequenceBinary(value);
   }
 
+  public static int compareToLexicographic(Binary one, Binary other) {
```

Comment:
I think, `lexicographicCompare` would be a better naming.

Reply:
It's called `compareToLexicographic` to keep it in-line with base class's generic `compareTo`

Reply:
`compareTo` is usually used when `this` is to be compared to the argument. In case of a method comparing the two arguments to each other `compare` is preferred. 
I think, either `lexicographicCompare` or `compareLexicographically` would be better than the current one.

The naming of the package private methods that really compares `this` to the argument is less important to me but still, something like `lexicographicCompareTo` or `compareToLexicographically` would be better.